content/docs/pipelines/pipelines-overview.md

```diff
@@ -155,6 +155,54 @@ Receiver operating characteristics (ROC) curve:
   alt="ROC"
   class="mt-3 mb-3 p-3 border border-info rounded">
 
+## Architectural overview
+
+<img src="/docs/images/pipelines-architecture.png" 
+  alt="Pipelines architectural diagram"
+  class="mt-3 mb-3 p-3 border border-info rounded">
+
+At a high level, the execution of a pipeline proceeds as follows: 
+
+* **Python SDK**: You create components or specify a pipeline using the Kubeflow
+  Pipelines domain-specific language 
+  ([DSL](https://github.com/kubeflow/pipelines/tree/master/sdk/python/kfp/dsl)).
+* **DSL compiler**: The
+  [DSL compiler](https://github.com/kubeflow/pipelines/tree/master/sdk/python/kfp/compiler)
+  transforms your pipeline's Python code into a static configuration (YAML).
+* **Pipeline Service**: You call the Pipeline Service to create a
+  pipeline run from the static configuration. 
+* **Kubernetes resources**: The Pipeline Service calls the Kubernetes API
+  server to create the necessary Kubernetes resources
+  ([CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/))
+  to run the pipeline.   
+* **Orchestration controllers**: A set of orchestration controllers
+  execute the containers needed to complete the pipeline execution specified
+  by the Kubernetes resources
+  ([CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)).
+  The containers execute within Kubernetes Pods on virtual machines. An
+  example controller is the **[Argo
+  Workflow**](https://github.com/argoproj/argo) controller, which
+  orchestrates task-driven workflows. 
+* **Artifact storage**: The Pods store two kinds of data: they call a Data
+  Artifact Storage Service (Minio Server, Google Cloud Storage, etc. to store
+  data artifacts. They call a Metrics Artifact Storage Service (GCP Stackdriver,
+  on-prem equivalent, etc) to store metrics artifacts.
+* **Persistence agent and ML metadata**: The Pipeline Persistence Agent
+  watches the Kubernetes resources created by the Pipeline Service and
+  persists the state of these resources in the ML Metadata Service. The
+  Pipeline Persistence Agent records the set of containers that executed as
+  well as their inputs and outputs. The input/output consists of either
+  container parameters or data artifact URIs. 
+* **Garbage collection**: The Data Artifact Garbage Collector queries the
+  Machine Learning Metadata Service to delete artifacts from artifact storage
+  if the system has been configured to delete artifacts that meet a certain
+  condition (for example, a time-to-live (TTL) has expired). 
```

Comment:
does pipeline has GC already or this is what we should do eventually? AFAIK GC is not yet supported.

Reply:
Thanks! I've removed GC from the text and diagram.